compiler-core/src/type_/error.rs

```diff
@@ -384,6 +384,14 @@ pub enum EmptyListCheckKind {
     NonEmpty,
 }
 
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum LiteralCollectionKind {
+    List,
+    Tuple,
+    BitArray,
```

Comment:
There's a use case for bit arrays! You could be matching on segments that don't align with the segments you used for construction.

Reply:
You're right, I've removed the warning when matching on non empty literal bit arrays 👍